include/gridtools/stencil_composition/sid/sid_helpers.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include "../../meta.hpp"
+#include "concept.hpp"
+#include "delegate.hpp"
+#include "multi_shift.hpp"
+
+namespace gridtools {
+    namespace sid {
+        namespace sid_helpers_impl_ {
+            template <class Sid>
+            class shifted_sid : public delegate<Sid> {
+                sid::ptr_holder_type<Sid> m_origin;
+
+                friend GT_META_CALL(sid::ptr_holder_type, Sid) sid_get_origin(shifted_sid &obj) { return obj.m_origin; }
+
+              public:
+                template <class Offsets>
+                constexpr shifted_sid(Sid &&original_sid, Offsets &&offsets) noexcept
+                    : delegate<Sid>(std::forward<Sid>(original_sid)), m_origin{[this, &offsets]() {
+                          auto &&strides = sid::get_strides(this->impl());
+                          GT_META_CALL(sid::ptr_diff_type, Sid) ptr_offset{};
+                          multi_shift(ptr_offset, strides, offsets);
+                          return sid::get_origin(this->impl()) + ptr_offset;
+                      }()} {}
+            };
+        } // namespace sid_helpers_impl_
+
+        template <class Sid, class Offset>
+        sid_helpers_impl_::shifted_sid<Sid> shifted_sid(Sid &&sid, Offset &&offset) {
```

Comment:
`sid_helpers_impl_::shifted_sid<Sid>` -> `sid_helpers_impl_::shifted_sid<decay_t<Sid>>`

Explanation:
According to the C++ rules `Sid` is deduced to the (maybe const) reference if the actual argument is an l-value.
The intent here is instantiate `sid_helpers_impl_::shifted_sid<Sid>` with non reference type in order to pass the ownership